### PR TITLE
Enhance Find feature

### LINKED ISSUE #63: Enhance find feature
Allow the list of clients to be filtered by attributes other than name.

### PR DESCRIPTION
**Pull request by cindyangXQ**
Allows finding keywords in each or all of the attributes. 
Command `find` changed to `find ATTR KEYWORD [keyword]...`, where `ATTR` can be one of:
`all` `name` `phone` `email` `address` `properties` `preference` `usertype`

Todo:
- [X] Implementation
- [X] Testing
- [X] Documentation

Closes #63.

### CODE REVIEW

File: src/test/java/seedu/address/testutil/TypicalPersons.java [hunk 1]
```diff
@@ -60,6 +60,24 @@
             .withEmail(VALID_EMAIL_BOB).withAddress(VALID_ADDRESS_BOB).withUserType(VALID_USERTYPE_BUYER)
             .withPreference(VALID_PREFERENCE_BOB).build();
 
+    // Sellers with properties
+    public static final Person PROTY = new PersonBuilder().withName("Proty Per").withPhone("35748962")
+            .withEmail("hastwo@property.com").withAddress("orchard").withUserType("seller")
```
Comment (hsiaojietng) [inline src/test/java/seedu/address/testutil/TypicalPersons.java:65]: Maybe change the address to a more likely one like "311, Clementi Ave 2, #02-25" (with street name, unit no).

**Review by hsiaojietng — APPROVED** [review]
LGTM, just some nits to check

File: src/main/java/seedu/address/model/predicates/PreferenceContainsKeywordsPredicate.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    public PreferenceContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        if (person.getPreference().isEmpty()) {
+            return false;
+        } else {
+            String preference = person.getPreference().get().toPlainString();
```
[the remark this answers is not in the document]
Comment (cindyangXQ) [reply src/main/java/seedu/address/model/predicates/PreferenceContainsKeywordsPredicate.java:27]: Do you mean line 27 only or this whole `test` method?

File: src/test/java/seedu/address/testutil/TypicalPersons.java (hunk 1 above)
Comment (cindyangXQ) [reply src/test/java/seedu/address/testutil/TypicalPersons.java]: Changes made!

File: src/main/java/seedu/address/model/predicates/PreferenceContainsKeywordsPredicate.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (hsiaojietng) [reply src/main/java/seedu/address/model/predicates/PreferenceContainsKeywordsPredicate.java:27]: Just this line 27 if possible

[the remark this answers is not in the document]
Comment (cindyangXQ) [reply src/main/java/seedu/address/model/predicates/PreferenceContainsKeywordsPredicate.java]: Changes made!

**Review by Ashuh — APPROVED** [review]
LGTM

**Pull request merged**